packages/fast-figma-plugin-msft/src/core/node.ts

```diff
@@ -0,0 +1,128 @@
+import { DesignSystem } from "@microsoft/fast-components-styles-msft";
+import { RecipeData, RecipeTypes } from "./recipe-registry";
+import { ColorRGBA64 } from "@microsoft/fast-colors";
+/**
+ * Defines the data stored by the plugin on a node instance
+ */
+export interface PluginNodeData {
+    /**
+     * A set of recipe IDs applied to the node
+     */
+    recipes: string[];
+
+    /**
+     * Design system overrides applied to the node
+     */
+    designSystem: Partial<DesignSystem>;
+}
+
+const DesignSystemCache: Map<string, Partial<DesignSystem>> = new Map();
+
+/**
+ * The abstract class the plugin controller interacts with
+ * Implementation details of this class will need to be created
+ * for each design tool.
+ */
+export abstract class PluginNode {
+    public abstract id: string;
+    public abstract type: string;
+    public abstract children(): PluginNode[];
+    public abstract parent(): PluginNode | null;
+    public abstract supports(): Array<RecipeTypes | "designSystem">;
+
+    /**
+     * Retrieves the contextual design system for the node
+     */
+    public get designSystem(): Partial<DesignSystem> {
+        // Retrun value from the cache if we have it
+        if (DesignSystemCache.has(this.id)) {
+            return DesignSystemCache.get(this.id) as Partial<DesignSystem>;
+        }
+
+        let node: PluginNode | null = this;
+        const designSystems: Array<Partial<DesignSystem>> = [];
+
+        while (node !== null) {
+            designSystems.push(node.getPluginData("designSystem"));
+            node = node.parent();
+        }
+
+        const designSystem = designSystems.reduceRight((prev, next) => ({
+            ...prev,
+            ...next,
+        }));
+        DesignSystemCache.set(this.id, designSystem);
+
+        return designSystem;
+    }
+
+    public get recipes(): string[] {
+        const recipes = this.getPluginData("recipes");
+        return recipes;
+    }
+
+    public set recipes(recipes: string[]) {
+        this.setPluginData("recipes", recipes);
+    }
+
+    /**
+     * Set a property of the design system on this node
+     * @param key - the design system property name
+     * @param value - the design system property value
+     */
+    public setDesignSystemProperty<K extends keyof DesignSystem>(
+        key: K,
+        value: DesignSystem[K]
+    ): void {
+        this.setPluginData("designSystem", {
+            ...this.getPluginData("designSystem"),
+            [key]: value,
+        });
+
+        this.invalidateDesignSystemCache();
+    }
+
+    public deleteDesignSystemProperty<K extends keyof DesignSystem>(key: K): void {

```

Comment:
Since you have a comment above on the set, should you add a code comment above the delete?